docs/aws/audit/esmonitoring/rules/es_whitelisted_ips_only_remediation.mdx

```diff
@@ -1,6 +1,254 @@
+    AccessPolicies=json.dumps(access_policy)
+)
+
+print("Access policy updated:", response)
+```
+
+### Key Points:
+1. **Initialize Boto3 Session**: Ensure you have the correct AWS credentials and region.
+2. **Create Elasticsearch Client**: Use the session to create an Elasticsearch client.
+3. **Define Access Policy**: Create a JSON policy that restricts access to specific IP addresses.
+4. **Update Domain Access Policy**: Apply the policy to your Elasticsearch domain using the `update_elasticsearch_domain_config` method.
+
+This script will help you ensure that your Elasticsearch domain is only accessible from the whitelisted IP addresses, thereby preventing unauthorized access.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Elasticsearch Service dashboard. 
+3. In the navigation pane, choose "Domains".
+4. Select the Elasticsearch domain that you want to examine. In the "Access Policy" section, check if the policy allows access from IP addresses other than the ones that are whitelisted. If the policy allows access from any IP address (0.0.0.0/0), then the selected Elasticsearch domain is accessible from any IP, which is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI. Make sure you have the necessary permissions to access the Elasticsearch domains.
+
+2. List all the Elasticsearch domains using the following command:
+```
+aws es list-domain-names
+```
+This command will return a list of all Elasticsearch domain names.
+
+3. For each domain, you can describe the Elasticsearch domain configuration using the following command:
+```
+aws es describe-elasticsearch-domain-config --domain-name <your-domain-name>
+```
+Replace `<your-domain-name>` with the name of your Elasticsearch domain. This command will return the configuration information of the specified Elasticsearch domain.
+
+4. Check the `AccessPolicies` in the output of the above command. This policy defines who can access the Elasticsearch domain and their permissions. If the `AccessPolicies` allows access from IP addresses that are not in your whitelist, then the Elasticsearch domain is misconfigured. The `AccessPolicies` should only allow access from whitelisted IP addresses.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries and establish a connection with AWS using Boto3 in Python. Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+
+# Create an AWS session
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>